website/docs/feature-flag-tutorials/react/examples.md

```diff
@@ -0,0 +1,627 @@
+
+
+Adjust the percentage of users to 50% or whichever value you choose, and refresh your app in the browser to see if your user has the new feature experience.
+
+
+![Gradual rollout form](/img/react-ex-grad-rollout-form.png)
+
+
+You can achieve this same result using our API with the following code: 
+
+```
+curl --location --request PUT 'http://localhost:4242/api/admin/projects/default/features/newNotificationsBadge/environments/development/strategies/{STRATEGY_ID}' \
+    --header 'Authorization: INSERT_API_KEY' \
+    --header 'Content-Type: application/json' \
+    --data-raw '{
+  "name": "flexibleRollout",
+  "title": "",
+  "constraints": [],
+  "parameters": {
+    "rollout": "50",
+    "stickiness": "default",
+    "groupId": "newNotificationsBadge"
+  },
+  "variants": [],
+  "segments": [],
+  "disabled": false
+}'
+
+```
+
+Learn more about [gradual rollouts in our docs](/reference/activation-strategies.md).
+
+
+## Canary Deployments in React
+
+
+### What is a canary deployment?
+
+Canary releases are a way to test and release code in different environments for a subset of your audience, which determines which features or versions of the platform people have access to.
+
+
+### Why use canary deployments?
+
+Canary deployments are a safer and more gradual way to make changes in software development. They help find any abnormalities and align with the agile process for faster releases and quick reversions.
+
+
+### How to leverage feature flags for canary deployments in React?
+
+Unleash has a few ways to help manage canary deployments for React apps at scale:
+
+
+* Using a [gradual rollout](/reference/activation-strategies#gradual-rollout) (which we implemented in a previous section) would be a simple use case but would reduce the amount of control you have over who gets the new feature.
+
+* Using either [constraints](/reference/strategy-constraints) or [segments](/reference/segments) (which are a collection of constraints) for a subset of your users to get the new feature vs. the old feature, for _more_ control than a gradual rollout
+
+* [Strategy variants](/reference/strategy-variants) are used to do the same canary deployment, but can be scaled to more _advanced_ cases. For example, if you have 2+ new features and are testing to see if they are better than the old one, you can use variants to split your population of users and conduct an A/B test with them.
+
+Let’s walk through how to utilize strategy constraints in our React app through the Unleash platform.
+
+
+#### Configure strategy constraints for canary deployments
+
+We will build a strategy constraint on top of our existing gradual rollout strategy. This will allow us to target a subset of users to rollout to.
+
+In Unleash, start from the [feature flag view](http://localhost:4242/projects/default/features/newNotificationsBadge) and edit your Gradual Rollout strategy from your development environment.
```

Comment:
I'm wondering about this link. It'll only work if the user has Unleash running on port 4242 and have the same setup as you. But maybe that's intentional? Only reading through, I expected it to go to something I could look at. But maybe it's intended to be read primarily by users who have followed the previous tutorial?